### PR TITLE
Upgrade to reason-js master

### PR DESCRIPTION
**Pull request by glennsl**
To be merged when ready

### CODE REVIEW

File: examples/simple/root.re [hunk 1]
```diff
@@ -1,3 +1,8 @@
 Js.log "Do you see this message? If you do, then things should work!";
 
-ReactDOMRe.render <Page message="Hello!" /> (ReasonJs.Document.getElementById "index");
+open ReasonJs;
+
+switch (document |> Document.getElementById "index") {
```
Comment (chenglou) [inline examples/simple/root.re:5]: Hummm... this is heavy, but that's the right way of doing it, I guess. Hopefully @bobzhang null-to-optional optimization works here!

Also, raising `Not_found` is right but might not be ideal for newcomers. `Invalid_argument "Index not found"` might help more.

File: examples/todomvc/app.re [hunk 2]
```diff
@@ -21,9 +23,9 @@ module Top = {
     };
     let getInitialState _ /* props */ => {
       let todos =
-        switch (Js.Null.to_opt (ReasonJs.LocalStorage.getItem namespace)) {
+        switch (ReasonJs.LocalStorage.getItem namespace) {
         | None => []
-        | Some todos => ReasonJs.JSON.parse todos
+        | Some todos => ReasonJs.Json.parse todos
```
Comment (chenglou) [inline examples/todomvc/app.re]: No need for `ReasonJs` here either then

File: examples/todomvc/app.re [hunk 3]
```diff
@@ -45,7 +50,7 @@ module Top = {
         | nonEmptyValue =>
           let todos =
             state.todos @ [
-              {id: string_of_float (ReasonJs.Date.now ()), title: nonEmptyValue, completed: false}
+              {id: string_of_float (Date.now ()), title: nonEmptyValue, completed: false}
```
Comment (chenglou) [inline examples/todomvc/app.re]: We have Js_date now right? Kill ReasonJs.Date! =D

File: examples/todomvc/todoItem.re [hunk 4]
```diff
@@ -45,7 +47,10 @@ module TodoItem = {
       None
     };
   let handleChange {props} event =>
-    props.editing ? Some {editText: ReasonJs.Document.value event##target} : None;
+    switch (ReasonJs.Element.asHtmlElement event##target) {
```
Comment (chenglou) [inline examples/todomvc/todoItem.re:48]: If it's just one usage, probably shouldn't open.

File: examples/simple/root.re (hunk 1 above)
Comment (glennsl) [reply examples/simple/root.re:5]: It's nicer with an Option module. I tend to use rebase, but I don't think that's a dependency that ought to be brought in here.

**Comment by glennsl** [comment]
Changes made according to your comments.

**Comment by glennsl** [comment]
Closing in favor of #61

**Pull request closed without merging**